core/templates/dev/head/pages/exploration_player/ProgressNavDirective.js

```diff
@@ -23,20 +23,41 @@ oppia.directive('progressNav', [
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/exploration_player/' +
         'progress_nav_directive.html'),
+      scope: {
+        submitFn: '&',
+        onClickContinueButton: '&'
+      },
       controller: [
         '$scope', '$rootScope', 'PlayerPositionService',
-        'PlayerTranscriptService',
+        'PlayerTranscriptService', 'ExplorationPlayerService',
+        'ExplorationPlayerStateService', 'CONTINUE_BUTTON_FOCUS_LABEL',
         function($scope, $rootScope, PlayerPositionService,
-          PlayerTranscriptService) {
-          $scope.currentCardIndex = PlayerPositionService.getActiveCardIndex();
-          $scope.transcriptLength = PlayerTranscriptService.getNumCards();
+          PlayerTranscriptService, ExplorationPlayerService,
+          ExplorationPlayerStateService, CONTINUE_BUTTON_FOCUS_LABEL) {
+          $scope.CONTINUE_BUTTON_FOCUS_LABEL = CONTINUE_BUTTON_FOCUS_LABEL;
+
+          var update = function() {
+            var index = PlayerPositionService.getActiveCardIndex();
+            $scope.activeCardIndex = index;
```

Comment:
Why not just
```
    $scope.activeCardIndex = (
      PlayerPositionService.getActiveCardIndex());
```
(and drop the index var)?

Reply:
Ah this was purely cosmetic because I use `index` later and it saves characters.

Reply:
Ah ok. Mildly prefer what I suggested, for two reasons:
- Fewer different vars to keep in the reader's head
- "index" on its own isn't a very descriptive variable name.

Reply:
Fixed.